x/programs/rust/wasmlanche-sdk/src/program.rs

```diff
@@ -6,11 +6,35 @@ use crate::{
     Gas,
 };
 use borsh::{BorshDeserialize, BorshSerialize};
-use std::{cell::RefCell, collections::HashMap};
+use std::{cell::RefCell, collections::HashMap, io::Read};
 use thiserror::Error;
 
+/// Defer deserialization from bytes
+#[cfg_attr(feature = "debug", derive(Debug))]
+pub struct DeferDeserialize(Vec<u8>);
+
+impl DeferDeserialize {
+    /// # Errors
+    /// Returns a [`std::io::Error`] if there was an issue deserializing the value
+    #[allow(clippy::must_use_candidate)]
+    pub fn deserialize<T: BorshDeserialize>(self) -> Result<T, std::io::Error> {
+        let Self(bytes) = self;
+        borsh::from_slice(&bytes)
+    }
+}
+
+impl BorshDeserialize for DeferDeserialize {
+    /// <div class="warning">This function does not provide any size-hint.
+    /// It is possible that this type performs multiple allocations during deserialization. It should be used sparingly.</div>
+    fn deserialize_reader<R: Read>(reader: &mut R) -> std::io::Result<Self> {
+        let mut inner = Vec::new();
+        reader.read_to_end(&mut inner)?;
+        Ok(Self(inner))
+    }
+}
+
 /// An error that is returned from call to public functions.
-#[derive(Error, Debug, BorshSerialize, BorshDeserialize)]
+#[derive(Error, Debug, BorshSerialize, BorshDeserialize, PartialEq, Eq)]
```

Comment:
```suggestion
#[derive(Error, Debug, BorshSerialize, BorshDeserialize)]
```

Reply:
I had to do that for the test `defer_err_variant`. It would not only be useful for the test but also for pattern-matching the error in programs directly, don't you think that we should leave it ?

Reply:
Now the test is gone, but the usage of testing if this error matches is still useful in programs context.

Reply:
No one should be comparing the internals of this type unless it's a test here which can do the comparison with `deferred.0`.

Reply:
Sorry, above comment doesn't make sense